src/ood-gen/lib/workshop.ml

```diff
@@ -0,0 +1,213 @@
+type role = [ `Chair | `Co_chair ]
+
+let role_to_string = function `Chair -> "chair" | `Co_chair -> "co-chair"
+
+let role_of_string = function
+  | "chair" -> Ok `Chair
+  | "co-chair" -> Ok `Co_chair
+  | _ -> Error (`Msg "Unknown role type")
+
+let role_of_yaml = function
+  | `String s -> Result.bind (role_of_string s) (fun t -> Ok t)
+  | _ -> Error (`Msg "Expected a string for a role type")
+
+let role_to_yaml t = `String (role_to_string t)
+
+type important_date = { date : string; info : string } [@@deriving yaml]
+
+type committee_member = {
+  name : string;
+  role : role option;
+  affiliation : string option;
+}
+[@@deriving yaml]
+
+type presentation = {
+  title : string;
+  authors : string list;
+  link : string option;
+  video : string option;
+  slides : string option;
+  additional_links : string list option;
+}
+[@@deriving yaml]
+
+type metadata = {
+  title : string;
+  location : string option;
+  date : string;
+  online : bool;
+  important_dates : important_date list;
+  presentations : presentation list;
+  program_committee : committee_member list;
+  organising_committee : committee_member list;
+}
+[@@deriving yaml]
+
+type t = {
+  title : string;
+  slug : string;
+  location : string option;
+  date : string;
+  online : bool;
+  important_dates : important_date list;
+  presentations : presentation list;
+  program_committee : committee_member list;
+  organising_committee : committee_member list;
+  toc_html : string;
+  body_md : string;
+  body_html : string;
+}
+
+let all () =
+  Utils.map_files
+    (fun content ->
+      let metadata, body = Utils.extract_metadata_body content in
+      let metadata = Utils.decode_or_raise metadata_of_yaml metadata in
+      let omd = Omd.of_string body in
+      {
+        title = metadata.title;
+        slug = Utils.slugify metadata.title;
+        location = metadata.location;
+        date = metadata.date;
+        online = metadata.online;
+        important_dates = metadata.important_dates;
+        presentations = metadata.presentations;
+        program_committee = metadata.program_committee;
+        organising_committee = metadata.organising_committee;
+        toc_html = Omd.to_html (Omd.toc omd);
+        body_md = body;
+        body_html = Omd.to_html omd;
+      })
+    "workshops/en/*.md"
```

Comment:
Is there a need to have the workshops translated one day?

Reply:
Yep you are right